pkg/manifests/manifests.go

```diff
@@ -26,20 +37,37 @@ func NewFactory(namespace string) *Factory {
 	}
 }
 
-func (f *Factory) PrometheusK8sRules() (*monv1.PrometheusRule, error) {
-	r, err := f.NewPrometheusRule(MustAssetReader(CephRules))
+func GetManifestsFromConfig() error {
```

Comment:
Can you make the above function something like this,
```go
func GetManifestsFromConfig() (map[string]map[string]string, error) {
    var rules map[string]map[string]string // this will be nill by default
    ...
    ...
    return rules, err
```
This way the function will be safeguarded from any side-effects (that means, it will not modify any global variables), when it is called from outside. On the call-site (line#56), you can do,
```go
rules, err := GetManifestsFromConfig()
```
You can avoid `rules` global variable all together.

PS: you have to change your tests as well.

Reply:
ack.